### PR TITLE
Add and for absolute indexed addressing

### PR DESCRIPTION
**Pull request by johnor**

### CODE REVIEW

File: core/test/src/test_cpu.cpp [hunk 1]
```diff
@@ -37,6 +37,8 @@ enum Opcode : uint8_t {
     AND_ABS = 0x2D,
     BMI = 0x30,
     SEC = 0x38,
+    AND_ABSY = 0x39,
```
Comment (robinlinden) [inline core/test/src/test_cpu.cpp]: AND_ABSY is missing tests.

Comment (johnor) [reply core/test/src/test_cpu.cpp]: Oh, right. Fixed!

File: core/src/mos6502.h [hunk 2]
```diff
@@ -80,12 +84,14 @@ class Mos6502 : public ICpu {
     Pipeline create_load_instruction(Opcode opcode);
     Pipeline create_compare_instruction(Opcode opcode);
 
-    Pipeline create_addressing_steps(AddressMode address_mode);
+    Pipeline create_addressing_steps(AddressMode address_mode,
+            bool is_write = false);
```
Comment (johnor) [inline core/src/mos6502.h]: Not sure what makes most sense here: default to write, default to read, or require it to be specified (in that case it should perhaps belong in the opcode struct?).

Comment (robinlinden) [reply core/src/mos6502.h]: I think this is good for now. We can change it in a separate PR if we come up with something nicer.

**Pull request merged**